html/rendering/widgets/baseline-alignment-and-overflow.tentative.html

```diff
@@ -0,0 +1,181 @@
+      'image',
+      'image-with-src',
+      'reset',
+      'button',
+    ];
+    for (let inputType of inputTypes) {
+      let src = "data:,broken";
+      if (inputType == "image-with-src") {
+        inputType = "image";
+        src = imageURL;
+      }
+      let inputAtts;
+      if (inputType === 'image') {
+        inputAtts = `src="${src}" alt="x"`;
+      } else {
+        inputAtts = `value="x"`;
+      }
+      for (var appearanceValue of ["auto", "none"]) {
+        for (var overflowValue of ['visible', 'hidden', 'scroll']) {
+          const tr = document.createElement('tr');
+          tr.innerHTML = `<td><span>${inputType}</span> <input type=${inputType} ${inputAtts} style="overflow: ${overflowValue}; appearance: ${appearanceValue};"></td>`;
+          testTable.append(tr);
+        }
+      }
+    }
+
+    // wait for images to load
+    await new Promise(resolve => window.onload = e => resolve());
+    for (const img of document.images) {
+      assert_precondition(img.complete); // either error state or loaded
+    }
+
+    // get layout info from refs
+    const refTextOffsetTop = document.querySelector('.ref-text span').offsetTop;
+    const refCheckboxOffsetTop = document.querySelector('.ref-checkbox span').offsetTop;
+    const refColorOffsetTop = document.querySelector('.ref-color span').offsetTop;
+    const refFileOffsetTop = document.querySelector('.ref-file span').offsetTop;
+    const refImageAltOffsetTop = document.querySelector('.ref-image-alt span').offsetTop;
+    const refImageSrcOffsetTop = document.querySelector('.ref-image-src span').offsetTop;
+
+    function expectedOffsetTop(input) {
+      // TODO(zcorpan) https://github.com/whatwg/html/issues/5065
+      // for now this is intended to match Firefox
+      const style = input.getAttribute('style');
+      const src = input.getAttribute('src');
+      let rv;
+      switch (input.type) {
+        case 'file':
+          rv = refFileOffsetTop; break;
+        case 'range':
+          rv = refImageSrcOffsetTop; break;
+        case 'color':
+          rv = refColorOffsetTop; break;
+        case 'checkbox':
+        case 'radio':
+          if (style.includes('appearance: none;')) {
+            rv = refImageSrcOffsetTop;
+          } else {
+            rv = refCheckboxOffsetTop;
+          }
+          break;
+        case 'file':
+          rv = refFileOffsetTop; break;
+        case 'image':
+          if (src === 'data:,broken' && style.includes('overflow: visible;')) {
+            rv = refImageAltOffsetTop;
+          } else {
+            rv = refImageSrcOffsetTop;
+          }
+          break;
+        default:
+          rv = refTextOffsetTop; break;
+      }
+      return rv;
+    }
+
+    function testName(markup) {
+      return markup.replace(imageURL, 'data:(png)');
+    }
+
+    for (const row of testTable.children) {
+      const input = row.firstChild.lastElementChild;
+      const allowedDelta = 3;
+      promise_test(async () => {
```

Comment:
Why is this a promise_test?

Reply:
Because with `promise_setup()` only `promise_test()` is allowed (see step 6 in https://github.com/web-platform-tests/rfcs/blob/master/rfcs/asynchronous_setup.md ).

Reply:
Could you add a comment explaining that?

Reply:
Done